src/participant/participant.ts

```diff
@@ -811,50 +784,205 @@ export default class ParticipantController {
+      )
+    );
+
+    this.renderCollectionsTree({
+      collections,
+      command,
+      databaseName,
+      context,
+      stream,
+    });
+  }
+
+  /** Gets the database name if there is only one collection.
+   *  Otherwise returns undefined and asks the user to select the database. */
+  async _getOrAskForDatabaseName({
+    command,
+    context,
+    stream,
+  }: {
+    command: ParticipantCommand;
+    context: vscode.ChatContext;
+    stream: vscode.ChatResponseStream;
+  }): Promise<string | undefined> {
+    const databases = await this._getDatabases({ stream });
+
+    if (databases === undefined || databases.length === 0) {
+      log.error('No databases found');
+      return;
+    }
+
+    if (databases.length === 1) {
+      return databases[0].name;
+    }
+
     // If no database or collection name is found in the user prompt,
     // we retrieve the available namespaces from the current connection.
     // Users can then select a value by clicking on an item in the list.
+    stream.markdown(
+      `What is the name of the database you would like${
+        command === '/query' ? ' this query' : ''
+      } to run against?\n\n`
+    );
+
+    this.renderDatabasesTree({
+      databases,
+      command,
+      context,
+      stream,
+    });
+  }
+
+  /** Helper which either automatically picks and returns missing parts of the namespace (if any)
+   *  or prompts the user to pick the missing namespace.
+   */
+  async _getOrAskForMissingNamespace({
+    databaseName,
+    collectionName,
+    context,
+    stream,
+    command,
+  }: {
+    databaseName: string | undefined;
+    collectionName: string | undefined;
+    context: vscode.ChatContext;
+    stream: vscode.ChatResponseStream;
+    command: ParticipantCommand;
+  }): Promise<{
+    databaseName: string | undefined;
+    collectionName: string | undefined;
+  }> {
     if (!databaseName) {
-      stream.markdown(
-        `What is the name of the database you would like${
-          command === '/query' ? ' this query' : ''
-        } to run against?\n\n`
-      );
-      await this.renderDatabasesTree({
+      databaseName = await this._getOrAskForDatabaseName({
         command,
         context,
         stream,
       });
-    } else if (!collectionName) {
-      stream.markdown(
-        `Which collection would you like to use within ${databaseName}?\n\n`
+
+      // If the database name could not get automatically selected,
+      // then the user has been prompted for it instead.
+      if (!databaseName) {
+        return { databaseName, collectionName };
+      }
+
+      // Save the database name in the metadata.
+      const chatId = ChatMetadataStore.getChatIdFromHistoryOrNewChatId(
+        context.history
       );
-      await this.renderCollectionsTree({
-        command,
+      this._chatMetadataStore.setChatMetadata(chatId, {
+        ...this._chatMetadataStore.getChatMetadata(chatId),
         databaseName,
+      });
```

Comment:
[nit] This is a bit repetitive - not sure if there are other instances beyond the db and collection names, but I'd consider extracting a method in ChatMetadataStore like:

```ts
patchChatMetadata(context: vscode.ChatContext, patchedMetadata: Partial<ChatMetadata>) {
  const chatId = getChatIdFromHistoryOrNewChatId(
    context.history
  );

  this.setChatMetadata(chatId, {
    ...this.getChatMetadata(chatId),
    ...patchedMetadata
  });
}
```